### DISCUSSION TITLE
RP2040 support

### DISCUSSION THREAD

**Discussion by mattbrailsford** [djyt/cannonball#133] [Ideas]
Hey folks,

I recently read an article in MagPi magazine where they got Doom running on the new Raspberry Pi RP2040 (Pico) boards and as a bit of a hardware hacker that likes to make mini arcades, I thought it would be amazing if cannonball was able to run on one of these.

Does anyone know if this would be possible? The blog post by the chap that got doom running (https://kilograham.github.io/rp2040-doom/) looks way over my head, but is there any reason cannonball couldn't be made to run on it?

**Comment by J1mbo** [comment]
The obvious issue is lack of RAM and lack of flash on the Pico.

The arcade machine used 2x 68000 CPUs each with 32KB RAM so it could presumably do the job of those and likely the sound (run by a Z80). 2MB flash is probably enough for that too.

But the video output is going to be challenging. There was a lot of hardware in the arcade board and we’d need a couple of frame buffers. Even if we used a second Pico as a pseudo video card somehow, it has enough ROM and maybe enough oomph but only enough RAM for a single frame buffer and even then assuming the code on it was tiny.

**Comment by J1mbo** [comment]
If you're still looking at this @mattbrailsford, my fork of this project runs on a Pi Zero 2W at 60fps and does so with CRT filtering: https://github.com/J1mbo/cannonball/

**Reply by mattbrailsford** [reply]
Ooooh, sweeeeeet! I'll defo give that a try.
